paf-mvp-core-js/src/express/node.ts

```diff
@@ -111,11 +111,19 @@ export class Node implements INode {
 
       // In case of timeout redirect to referer ...
       if (err.message === 'Response timeout') {
+        // FIXME[errors] only in case of redirect
         const error: NodeError = {
           type: NodeErrorType.RESPONSE_TIMEOUT,
           details: err.message,
         };
         this.redirectWithError(res, req.header('referer'), 504, error);
+      } else if ((err as NodeError).type) {
```

Comment:
This will be refined: we will check each known `type`.
Today it's only to distinguish known errors from exception (since there is no type checking in JS)

Reply:
Then you can only check if it is a NodeError. It is simpler.

Reply:
How will you check if it is a `NodeError`?
> there is no type checking in JS

Reply:
We have [`instanceof`](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#instanceof-narrowing).

Reply:
No: it only works with classes, not with interfaces and even in this case, you need to do dirty hacks like https://github.com/OneKey-Network/OneKey-implementation/blob/b23a0b6e553e765af75bebecd764b7bd0512bddd/paf-mvp-core-js/src/express/errors.ts#L10 which I would **really** like to avoid.

Reply:
Ok thank you for the clarification.